hermeto/core/package_managers/javascript/yarn/main.py

```
@@ -110,37 +169,73 @@ def _verify_repository(project: Project) -> None:
-    _verify_repository(project)
+    git_deps = _parse_lockfile_git_deps(project)
 
-    _set_yarnrc_configuration(project, output_dir, version)
-
-    packages = resolve_packages(project.source_dir, workspaces)
+    # Reject before mutating yarnrc / the lockfile when git deps need permissive mode.
+    if git_deps and mode != Mode.PERMISSIVE:
+        raise PackageRejected(
+            "Git dependencies in Yarn Berry projects cannot be processed in strict mode "
+            "because the lockfile must be modified to reference local tarballs."
+        )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)